File: 21-merge-two-sorted-lists/21-merge-two-sorted-lists.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* mergeTwoLists(ListNode* list1, ListNode* list2) {
        ListNode* ans=new ListNode(0);
        ListNode* curr=ans;
        while(list1!=NULL && list2!=NULL){
            if(list1->val > list2->val){
                curr->next=list2;
                curr=curr->next;
                list2=list2->next;
            }
            else{
                curr->next=list1;
                curr=curr->next;
                list1=list1->next;
            }
        }
        while(list2!=NULL){
            curr->next=list2;
            curr=curr->next;
            list2=list2->next;
        }
        while(list1!=NULL){
            curr->next=list1;
            curr=curr->next;
            list1=list1->next;
        }
        return ans->next;
    }
};

File: 21-merge-two-sorted-lists/21-merge-two-sorted-lists.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode mergeTwoLists(ListNode l1, ListNode l2) {
        ListNode ans=new ListNode(-1);
        ListNode edit=ans;
        while(l1!=null && l2!=null){
            if(l1.val > l2.val){
                edit.next=l2;
                l2=l2.next;
            }
            else{
                edit.next=l1;
                l1=l1.next;
            }
            edit=edit.next;
        }
        if(l1!=null){
            edit.next=l1;
        }
        if(l2!=null){
            edit.next=l2;
        }
        return ans.next;
    }
}